components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -283,13 +351,21 @@ open class GleanInternalAPI internal constructor () {
 
         // The first time Glean runs, we set the client id and other internal
         // one-time only metrics.
-        val firstRunDetector = FileFirstRunDetector(gleanDataDir)
-        if (firstRunDetector.isFirstRun()) {
+        fixLegacyPingInfoMetrics()
+
+        val clientIdMetric = GleanInternalMetrics.clientId
+        val existingClientId = UuidsStorageEngine.getSnapshot(
+            clientIdMetric.sendInPings[0], false)?.get(clientIdMetric.identifier)
+        if (existingClientId == null || existingClientId == KNOWN_CLIENT_ID) {
             val uuid = UUID.randomUUID()
-            UuidsStorageEngine.record(GleanInternalMetrics.clientId, uuid)
-            DatetimesStorageEngine.set(GleanInternalMetrics.firstRunDate)
-        } else {
-            fixLegacyPingInfoMetrics()
+            UuidsStorageEngine.record(clientIdMetric, uuid)
+        }
+
+        val firstRunDateMetric = GleanInternalMetrics.firstRunDate
```

Comment:
Hah, so no more `FileFirstRunDetector`. I like that.

Reply:
Yep.  :bomb: